### PR TITLE
Use stderr by default when a specific output is not injected

### PR DESCRIPTION
**Pull request by Seldaek**
| Q | A |
| --- | --- |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets |  |
| License | MIT |
| Doc PR |  |

stderr is really the best place to put logs by default, so unless someone explicitly passes something else we should use it, and especially not depend on the log level to decide where to output.

### CODE REVIEW

**Comment by jakzal** [comment]
This change broke some monolog bridge tests.

**Comment by Seldaek** [comment]
Sorry, fixed the tests.

**Comment by Tobion** [comment]
Related #13730

**Comment by Seldaek** [comment]
@Tobion in retrospect, I think that other one you linked was misguided, it should be given the right output by the user instead of doing automagic. But in this case there is no easy way for the user to do it right, so I think defaulting to the right behavior when setup with the listener is good, and if people want to do it wrong they can always reconfigure the hard way.

File: src/Symfony/Bridge/Monolog/Handler/ConsoleHandler.php [hunk 1]
```diff
@@ -120,7 +120,11 @@ public function close()
      */
     public function onCommand(ConsoleCommandEvent $event)
     {
-        $this->setOutput($event->getOutput());
```
Comment (hhamon) [inline src/Symfony/Bridge/Monolog/Handler/ConsoleHandler.php]: For convenience, you could use a temporary variable:

    ```php
    $output = $event->getOutput();
    if ($output instanceof ConsoleOutputInterface) {
        $output = $output->getErrorOutput();
    }

    $this->setOutput($outpout);

Comment (Seldaek) [inline src/Symfony/Bridge/Monolog/Handler/ConsoleHandler.php]: I guess that's a bit nicer yes, fixed.

**Comment by fabpot** [comment]
:+1:

File: src/Symfony/Bridge/Monolog/Tests/Handler/ConsoleHandlerTest.php [hunk 2]
```diff
@@ -110,31 +110,18 @@ public function testGetFormatter()
...
-        $output = $this->getMock('Symfony\Component\Console\Output\ConsoleOutputInterface');
+        $output = $this->getMock('Symfony\Component\Console\Output\OutputInterface');
         $output
             ->expects($this->any())
             ->method('getVerbosity')
             ->will($this->returnValue(OutputInterface::VERBOSITY_DEBUG))
         ;
         $output
-            ->expects($this->once())
+            ->expects($this->any())
```
Comment (Tobion) [inline src/Symfony/Bridge/Monolog/Tests/Handler/ConsoleHandlerTest.php]: Why did you change that? It should only be callec once.

Comment (Seldaek) [inline src/Symfony/Bridge/Monolog/Tests/Handler/ConsoleHandlerTest.php]: Ah that was trying to achieve something else and then I gave up and forgot to change it back. Let me change it back quickly.

**Comment by Tobion** [comment]
:+1:

**Comment by fabpot** [comment]
Thank you @Seldaek.
